### ISSUE TITLE
Types `object` and `Callable` are reported to not have any overlap.

### ISSUE THREAD

**Issue by ImogenBits** [microsoft/pyright#10253]
When using the reportUnnecessaryComparison option, the following code generates a warning because `object` and any callable type supposedly have no overlap.

```python
def f(a: object, b: Callable[..., int]) -> bool:
    return a is b
```

I'm using the VSCode extension with pyright version 1.1.398, the same behaviour also occurs in the [pyright playground](https://pyright-play.net/?pythonVersion=3.12&strict=true&code=GYJw9gtgBAxmA28CmMAuBLMA7AzgOgEMAjGKdCABzBFSgGEDFjkAoFgEyWCmAAoCAXFDBEAVilQAaKESEMmRZAG08q6eiyoAugEooAWgB8MsAgEsolqCCSoAriCxQCZHDKA)

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.399.
